flocker/provision/_install.py

```diff
@@ -72,15 +72,50 @@ def task_install_ssh_key():
     ])
 
 
-def task_upgrade_kernel():
+def task_upgrade_kernel(distribution):
     """
     Upgrade kernel.
     """
-    return sequence([
-        run_from_args(['yum', 'upgrade', '-y', 'kernel']),
-        comment(comment="# The upgrade doesn't make the new kernel default."),
-        run_from_args(['grubby', '--set-default-index', '0']),
-    ])
+    if distribution == 'fedora-20':
+        return sequence([
+            run_from_args(['yum', 'upgrade', '-y', 'kernel']),
+            comment(
+                comment="The upgrade doesn't make the new kernel default."),
+            run_from_args(['grubby', '--set-default-index', '0']),
+        ])
+    elif distribution == 'centos-7':
+        return sequence([
+            run_from_args([
+                "yum", "install", "-y", "kernel-devel", "kernel"]),
+            run_from_args(['sync']),
+        ])
+    elif distribution == 'ubuntu-14.04':
+        # When 15.04 is available then the kernel can be backported from that,
```

Comment:
This is due April 23rd. Follow up issue? Or if it'd be bad to ship this as-is maybe we should wait.